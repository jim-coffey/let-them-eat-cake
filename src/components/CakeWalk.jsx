var React = require('react');
var PropTypes = require('prop-types');

var Cake = require('./Cake');

const CakeWalk = (props) => {
  const { cakes, searchString, saveCake } = props;

  return (
    <ul className="cake-list">
      {
        cakes
          .filter(cake => {
            return cake.title.indexOf(searchString) > -1
                || cake.desc.indexOf(searchString) > -1;
          })
          .map((cake, index) => {
            return (
              <Cake key={cake.id} cake={cake} saveCake={saveCake} />
            );
          })
      }
    </ul>
  );
}

CakeWalk.propTypes = {
  cakes: PropTypes.array.isRequired,
  saveCake: PropTypes.func.isRequired,
  searchString: PropTypes.string
}

module.exports = CakeWalk;
