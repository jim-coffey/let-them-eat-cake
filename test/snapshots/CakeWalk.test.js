import React from 'react';
import renderer from 'react-test-renderer';

import CakeWalk from '../../src/components/CakeWalk';

describe('CakeWalk (Snapshot)', () => {
  it('renders correctly', () => {
    const cakes = [{
      id: 0,
      title: 'Cake1',
      desc: 'desc1',
      image: 'http://image'
    }]
    const saveCake = () => {};
    const searchString = '';

    const tree = renderer.create(
      <CakeWalk cakes={cakes} saveCake={saveCake} searchString={searchString} />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
