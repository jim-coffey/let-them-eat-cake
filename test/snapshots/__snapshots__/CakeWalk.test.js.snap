// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CakeWalk (Snapshot) renders correctly 1`] = `
<ul
  className="cake-list"
>
  <li
    className="cake-item"
  >
    <ul
      className="space-list-items"
    >
      <li>
        <a
          href="#"
          onClick={[Function]}
        >
          Edit
        </a>
      </li>
      <li>
        <img
          alt="Image for Cake1"
          className="cake-pic"
          src="http://image"
        />
      </li>
      <li
        className="header"
      >
        Cake1
      </li>
      <li
        className="desc"
      >
        desc1
      </li>
    </ul>
  </li>
</ul>
`;
